metadata/snapshot.go

```diff
@@ -302,14 +357,48 @@ func (s *snapshotter) createSnapshot(ctx context.Context, key, parent string, re
 	}
 
 	var (
-		target  = base.Labels[labelSnapshotRef]
-		bparent string
-		bkey    string
-		bopts   = []snapshots.Opt{
+		target               = base.Labels[labelSnapshotRef]
+		_, disableSameUnpack = base.Labels[labelDisableSameUnpack]
+		bparent              string
+		bkey                 string
+		bopts                = []snapshots.Opt{
 			snapshots.WithLabels(snapshots.FilterInheritedLabels(base.Labels)),
 		}
 	)
 
+	if disableSameUnpack {
+		bc, ok := s.inp.loadOrStore(key)
```

Comment:
If one client, A, starts unpacking a layer without `disableSameUnpack`, they wouldn't be registered in `inp`, so that if another client, B, comes in and unpacks that layer with `disableSameUnpack`, B would have no way of knowing about and waiting on A? 
So B will unpack the same layer again because A didnt opt in?

Reply:
Right, the way that this would be enabled would be a Containerd config option so the hope is that this isn't even a thing we need to worry about. We need to set some sort of data so the snapshotter knows we're asking for this and that's the label that you'd had a comment on above but this shouldn't be set by a client explicitly

Reply:
Glad you commented this because I forgot to push the cri changes needed (as our fork doesn't contain the cri code 😔)

Reply:
Here's how it would be turned on by a user in the containerd config